lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -348,4 +456,28 @@ public AggregatedMemoryContext getSystemMemoryContext()
     {
         return systemMemoryContext;
     }
+
+    private static <T> List<T> listWithNulls(int size)
+    {
+        return Stream.generate(() -> (T) null).limit(size).collect(Collectors.toCollection(ArrayList<T>::new));
+    }
+
+    private RowRanges getRowRanges(int blockIndex)
+    {
+        requireNonNull(filter, "filter is null");
```

Comment:
If the method cannot be called when `filter` is null, it would be cleaner to model it as an argument to the method that can never be null. The caller would be responsible for calling the method only when it makes sense to do so and the intention would be clear.

As it stands, the method has hidden behaviors that depend on the current state of the object in a non-obvious way.